### DISCUSSION TITLE
What happens when connection to db datastore is lost?

### DISCUSSION THREAD

**Discussion by RogerSik** [k3s-io/k3s#8034] [Q&A] [answered]
What happens when connection to db like postgresql datastore is lost?

Will every be fine until there is a change requested? Or will k3s go amok or or?

**Comment by brandond** [comment, marked as answer]
Agent nodes will continue to work fine without access to the apiserver, but the apiserver itself, and the core Kubernetes controllers, will all fail to renew or take the leader-election locks and crash, which will cause the K3s server process to restart and wait until the datastore becomes available. Pods on the server will continue to run, assuming they can function without access to the apiserve.
